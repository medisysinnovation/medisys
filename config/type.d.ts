export interface Network {
  endpoint: any;
}

export interface MedisysConfig {
  network: Network;
}
