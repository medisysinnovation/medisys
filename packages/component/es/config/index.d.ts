export declare type SizeType = 'small' | 'middle' | 'large' | undefined;
