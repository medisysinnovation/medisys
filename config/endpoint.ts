export default {
  login: "/connect/token",
  codetable: "/api/CodeTable",
  ctPort: "/api/CTPort",
  ctContainerOperator: "/api/CTContainerOperator",
  currentUser: "/api/User/Current",
  changePassword: "/api/User/ChangePassword",
  user: "/api/user",
  patient: "/api/patient",
  role: "/api/role",
  tenant: "/api/tenant",
  transporter: "/api/transporter",
  validateTransporter: "/api/Transporter/Validate",
  resetPassword: "/api/user/resetPassword",
  forgetPassword: "/api/user/forgetPassword",
  j5ApplicationForm: "/api/J5ApplicationForm",
  cancelApplication: "/api/J5ApplicationForm/Cancel",
  unNumberData: "/api/J5ApplicationForm/UNNumberData",
  j5Job: "/api/j5Job",
  initiateJob: "/api/j5Job/InitiateJob",
  cancelJob: "/api/j5Job/CancelJob",
  file: "/api/File",
  kdCargo: "/api/KDCargo",
  kioskJob: "/api/Kiosk",
  kioskReportIssue: "/api/Kiosk/ReportIssue",
  cargoLabel: "/api/Kiosk/cargoLabel",
  cargoDetails: "/api/CargoDetails",
  cargoDetailsAck: "/api/CargoDetails/Acknowledge",
  requestCargoImage: "/api/CargoDetails/RequestCargoImg",
  subscribeTopic: "/api/Message/SubscribeToTopic",
  unsubscribeTopic: "/api/Message/UnsubscribeFromTopic",
  notification: "/api/Notification",
  taskList: "/api/TaskList",
  approveContainerImage: "/api/J5ApplicationForm/ApproveContainerImg",
  approveCargoImage: "/api/CargoDetails/ApproveCargoImg",
  readAllNotification: "/api/Notification/ReadAll",
  requestContainerImage: "/api/J5ApplicationForm/RequestContainerImg",
  requestTransferOwnership: "/api/J5ApplicationForm/RequestTransferOwnership",
  approveTransferOwnership: "/api/J5ApplicationForm/ApproveTransferOwnership",
  validateCargoTransferOwnership:
    "/api/J5ApplicationForm/validateCargoTransferOwnership",
  systemSetting: "/api/SystemSetting",
  systemSettingExitKiosk: "/api/SystemSetting/ExitKiosk",
  systemSettingKioskHelper: "/api/SystemSetting/KioskHelperUrl",
  allSystemSetting: "/api/SystemSetting/All",
  storageList: "/api/StorageLayout/Storages",
  storageLayout: "/api/StorageLayout",
  storageOccupancy: "/api/StorageOccupancy",
  occupiedStorage: "/api/StorageOccupancy/Occupied",
  ctPublicHoliday: "/api/CTPublicHoliday",
  ctLabelPrinter: "/api/CTLabelPrinter",
  getJobImages: "/api/j5Job/images",
  validateImportSO: "/api/CargoDetails/ValidateImportSO",
  searchCargoDetails: "/api/CargoDetails/SearchCargo",
  ctUNNo: "/api/CTUNNo",
  unNoIMOClass: "/api/UNNoIMOClass",
  rptUNNoIMOClass: "/api/UNNumberReport",
  containerYard: "/api/ContainerYard",
  containerYardSummary: "/api/ContainerYard/Summary",
  inspectAll: "/api/ContainerMonitoring/InspectAll",
  containerMonitoring: "/api/ContainerMonitoring",
  updateSpecialDG: "/api/ContainerMonitoring/UpdateSpecialDG",
  updateContainerPhotos: "/api/ContainerMonitoring/UpdateContainerPhotos",
  getContainerReportData: "/api/ContainerMonitoring/ContainerReport",
  hauliers: "/api/Tenant/Hauliers",
  cargoSummary: "/api/CargoDetails/Summary",
  transferOwnership: "/api/J5ApplicationForm/TransferOwnership",
  getPM4Status: "/api/J5ApplicationForm/GetPM4Status",
  lotAllocationAck: "/api/CargoStoringOrderList/LotAllocationAck",
  dgSafetyInspectionReport: "/api/Report/DGSafetyInspection",
  sendDGSafetyInspection: "/api/Report/SendDGSafetyInspection",
  getInspectionAttachment: "/api/ContainerMonitoring/InspectionAttachment",
  getOtherLocationContainerMonitoring:
    "/api/ContainerMonitoring/OtherLocationContainerMonitoring",
  getYardTerminalGroup: "/api/ContainerMonitoring/GetYardTerminalGroup",
  getPastLocationAsync: "/api/ContainerMonitoring/GetPastLocation",
  getLatestContainerMonitoringData:
    "/api/ContainerMonitoring/GetLatestContainerMonitoringData",
  getByStoringOrderNumber: "/api/CargoDetails/GetByStoringOrderNumber",
  saveYardInspectionAttachment: "/api/ContainerYard/YardPhotos",
  downloadCargoCliqueReportTool: "/api/Download/CargoCliqueReportTool",
  report: "/api/report",
};
