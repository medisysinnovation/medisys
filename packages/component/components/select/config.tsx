export interface ConfigParamter {
  loader: (code: string) => void;
}

let _loader: Function = () => {};
const _config: ConfigParamter & any = {
  load: ({ code }: { code: string } = { code: '' }) => {
    console.log('load' + code);
    _loader({ code });
  },
};

_config.loader = (payload: ConfigParamter) => {
  _loader = payload.loader;
};
export default _config;
