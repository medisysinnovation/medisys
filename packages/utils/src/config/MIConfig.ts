import immutable from 'immutable';
console.log(immutable);
const { Map, List } = immutable;
export interface ModalStaticFunctions {
  updateState: (newState: StateProps) => void;
  loadData: (code: string, params?: any) => Promise<[]>;
}
// type d = typeof {[key: string]: V}
export interface StateProps {
  loading?: Object;
  dataSource?: Object;
}

export interface MedisysConfigProps {
  dataLoader?: ({ code, ...props }: { code: string }) => Promise<[]>;
  urls?: { [key: string]: string };
}

let localStore: StateProps = {
  loading: {},
  dataSource: JSON.parse(sessionStorage.getItem('mi_ds') || '{}'),
};

class MIConfig {
  static dataLoader = undefined;
  static urls = {
    login: '/connect/token',
    codetable: '/api/CodeTable',
    currentUser: '/api/User/Current',
    changePassword: '/api/User/ChangePassword',
    user: '/api/user',
    role: '/api/role',
    tenant: '/api/tenant',
  };
  static imt_current = immutable.fromJS(localStore);

  static config({ dataLoader, urls }: MedisysConfigProps) {
    //@ts-ignore
    if (dataLoader) this.dataLoader = dataLoader;

    if (urls) {
      this.urls = { ...this.urls, ...urls };
    }
  }

  static initialization() {
    var ds = this.imt_current.get('dataSource');

    ds.keySeq().forEach((code: string) => {
      // console.log(ds.get(code)?.toJS(), code);
      document.dispatchEvent(
        new CustomEvent('mi_datasourcechanged_' + code, {
          bubbles: true,
          detail: ds.get(code)?.toJS() ?? [],
        }),
      );
    });

    window.addEventListener('keydown', event => {
      if (event.ctrlKey && event.key === 'F5') {
        sessionStorage.removeItem('mi_ds');
        event.preventDefault();
        window.location.reload();
      }
    });
  }

  static getData(code: string) {
    if (!code) throw 'Must pass in `code`';
    return immutable.getIn(this.imt_current, ['dataSource', code], []).toJS();
  }

  static async loadData(code: string, params?: any) {
    // console.log(code);
    if (!this.dataLoader) {
      throw 'No default loader configed, please use `config` function set default dataLoader';
    }

    //@ts-ignore
    const data = await this.dataLoader({ code, ...params });
    if (data) {
      this.updateState({
        dataSource: {
          [code]: data,
        },
      });
    }

    return data;
  }

  static updateState(newState: StateProps = {}) {
    const { loading, dataSource } = newState;
    let imt_data = immutable.fromJS(newState) as Map<string, any>;
    if (
      loading &&
      imt_data.get('loading') !== this.imt_current.get('loading')
    ) {
      // console.log(loading);
      // console.log('loading state changed', loading, localStore.loading);
      document.dispatchEvent(
        new CustomEvent('loadingstatechanged', {
          bubbles: true,
          detail: loading,
        }),
      );
    }
    const dataSourceChanged =
      immutable.getIn(imt_data, ['dataSource'], Map()) !==
      immutable.getIn(this.imt_current, ['dataSource'], Map());
    // console.log(
    //   dataSourceChanged,
    //   immutable.getIn(imt_data, ['dataSource'], Map()).toJS(),
    //   immutable.getIn(this.imt_current, ['dataSource'], Map()).toJS(),
    //   immutable
    //     .merge(imt_data.get('dataSource'), this.imt_current.get('dataSource'))
    //     .toJS(),
    // );
    if (dataSource && dataSourceChanged) {
      let imt_dataSource = imt_data.get('dataSource');

      imt_dataSource.keySeq().forEach((code: string) => {
        document.dispatchEvent(
          new CustomEvent('mi_datasourcechanged_' + code, {
            bubbles: true,
            detail: imt_dataSource.get(code).toJS(),
          }),
        );
        // console.log(code);
      });

      imt_data = imt_data.set(
        'dataSource',
        immutable.merge(
          this.imt_current.get('dataSource'),
          imt_data.get('dataSource'),
        ),
      );
    }

    this.imt_current = immutable.merge(this.imt_current, imt_data);
    // console.log(imt_current.toJS());
    if (dataSourceChanged) {
      // console.log('set', imt_current.get('dataSource').toJS());
      sessionStorage.setItem(
        'mi_ds',
        JSON.stringify(this.imt_current.get('dataSource').toJS()),
      );
    }
  }
}

setTimeout(() => {
  MIConfig.initialization();
}, 0);

export default MIConfig;
