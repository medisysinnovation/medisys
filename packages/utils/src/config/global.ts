export interface ModalStaticFunctions {
  updateState: any;
}
export interface StateProps {
  loading: Object;
}

let localStore = {};

const global = {
  updateState: (newState: StateProps) => {
    localStore = newState;
    const { loading } = newState;
    // console.log(loading);
    document.dispatchEvent(
      new CustomEvent('loadingstatechanged', {
        bubbles: true,
        detail: loading,
      }),
    );
  },
};

export default global;
