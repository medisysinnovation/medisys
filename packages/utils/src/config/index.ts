import _ from 'lodash';
export { default as MIConfig } from './MIConfig';
export function defineConfig(config: Object) {
  return config;
}

export interface Network {
  endpoint: any;
}

export interface MIConfig {
  network: Network;
}
let config: MIConfig = {
  network: {
    endpoint: {},
  },
};

const initilaze = (_config: MIConfig) => {
  config = _.merge(config, _config);
};
// setInterval(() => {
//   console.log(config);
// }, 10000);

export { initilaze, config };
