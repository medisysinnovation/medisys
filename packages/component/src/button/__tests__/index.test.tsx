import React from 'react';
import { mount, render } from 'enzyme';
import mountTest from '../../../test/shared/mountTest';
import Button from '..';
import { SizeType } from '../../config';

describe('Button', () => {
  mountTest(Button);
  it('renders correctly', () => {
    expect(<Button>Follow</Button>).toMatchRenderedSnapshot();
  });
  it('mount correctly', () => {
    expect(() => mount(<Button>Follow</Button>)).not.toThrow();
  });
  it('warns if size is wrong', () => {
    const mockWarn = jest.fn();
    jest.spyOn(console, 'warn').mockImplementation(mockWarn);
    const size = ('who am I' as any) as SizeType;
    render(<Button.Group size={size} />);
    expect(mockWarn).toHaveBeenCalledTimes(1);
    expect(mockWarn.mock.calls[0][0]).toMatchObject({
      message: 'unreachable case: "who am I"',
    });
  });
});
