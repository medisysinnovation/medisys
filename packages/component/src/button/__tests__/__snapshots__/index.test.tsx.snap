// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
initialize {
  "0": Node {
    "attribs": Object {
      "class": "ant-btn",
      "type": "button",
    },
    "children": Array [
      Node {
        "attribs": Object {},
        "children": Array [
          Node {
            "data": "Follow",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "span",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
    ],
    "name": "button",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "type": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "type": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
